packages/flutter_blue_plus_platform_interface/lib/src/bluetooth_msgs.dart

```diff
@@ -318,13 +319,19 @@ class BmBluetoothCharacteristic {
     // convert descriptors
     List<BmBluetoothDescriptor> descs = [];
     for (var v in json['descriptors']) {
-      descs.add(BmBluetoothDescriptor.fromMap(v));
+      final Map<dynamic, dynamic> bv = {};
+      for (var key in v.keys) {
+        bv[key] = v[key];
+      }
+      bv['characteristic_index'] = json['characteristic_index'];
+      descs.add(BmBluetoothDescriptor.fromMap(bv));
     }
 
     return BmBluetoothCharacteristic(
       remoteId: DeviceIdentifier(json['remote_id']),
       serviceUuid: Guid(json['service_uuid']),
       characteristicUuid: Guid(json['characteristic_uuid']),
+      characteristicIndex: int.parse(json['characteristic_index']),
```

Comment:
Could we use an `int` instead of a `String` when serializing and deserializing the index?

```suggestion
      characteristicIndex: json['characteristic_index'],
```

https://docs.flutter.dev/platform-integration/platform-channels#codec